[RES-4518] Fix Orillusion z-order sort util when using a reversed depth buffer

stefco opened the pull request:
## Description

We added reversed depth buffer support to Orillusion, which replaces depth with 1 - depth to reduce z-fighting (since this is used in Wellington and we need to share depth buffers). However, we didn't update the z-sorter util used for things like enforcing transparency rendering order to account for this switch in ordering. Add support to worldToCameraDepth for reversed z-buffers to fix this.

## Issue

[RES-4518](https://linear.app/resolve/issue/RES-4518)

stefco committed:
add ZSorterUtil reversed depth buffer support

stefco committed:
fix reversed depth buffer usage in sample

stefco committed:
bump resolve release

jgilm commented on src/gfx/renderJob/collect/EntityCollect.ts:
```diff
@@ -278,9 +278,11 @@ export class EntityCollect {
-            });
+
+            const sortFn: (a: RenderNode, b: RenderNode) => number = Engine3D.setting.render.useReversedDepth
+                ? (a, b) => (a['__renderOrder'] - b['__renderOrder'] > 0 ? -1 : 1)
+                : (a, b) => (a['__renderOrder'] - b['__renderOrder'] > 0 ? 1 : -1);
```
You can simplify the sort functions, remove the ternary, and also cover the missing case of equals (0) just by using subtraction.

```suggestion
                ? (a, b) => b['__renderOrder'] - a['__renderOrder']
                : (a, b) => a['__renderOrder'] - b['__renderOrder'];
```

stefco replied:
I was thinking this too, but i wanted to match the existing style closely for when we submit this upstream

jgilm replied:
I gotcha. Maybe `__renderOrder` can never be equals between two RenderNode's, but if it ever is, then the sort will be unstable. Plus, branching for every comparison seems unoptimized, when it can be just handled by subtraction.

jgilm replied:
Final thought, I'll leave it to you, if it is changing a single line, I think changing the workings is acceptable since `sort` (I assume) is the built-in function, and the built-in function only compares negative, 0, and positive. Plus it makes it technically correct. I think you could state this as improving the code in addition to adding a feature. Sorry, one more thing, every comparison function I've ever seen using the subtraction method, so this would align with the vast majority of code in existence and thereby make it more easily parsable.

stefco replied:
yeah, i agree with all your points. i'll change it!

jgilm commented on src/util/ZSorterUtil.ts:
```diff
@@ -46,9 +47,11 @@ export class ZSorterUtil {
-            return a.z - b.z > 0 ? 1 : -1;
-        });
+        const sortFn: (a: { z: number }, b: { z: number }) => number = Engine3D.setting.render.useReversedDepth
+            ? (a, b) => (a.z - b.z > 0 ? -1 : 1)
+            : (a, b) => (a.z - b.z > 0 ? 1 : -1);
```
```suggestion
            ? (a, b) => b.z - a.z
            : (a, b) => a.z - b.z;
```

jgilm reviewed (approved):
Looks good, other than the comments I made.

stefco committed:
Use more idiomatic sorting

- avoiding a branch should be more efficient
- allowing 0 in comparison allows sort to be stable (implementation dependent)
- `sort` only distinguishes negative, 0, and positive
- this is how numeric comparison functions are usually defined

Co-authored-by: Jim Gilmartin <2531809+jgilm@users.noreply.github.com>
